test/access_control.sh

```diff
@@ -132,85 +162,98 @@ function test_user_has_proper_access() {
-
-            test_message="invalid access from kibana with no token"
-            os::cmd::expect_success "run_curl_cmd 403 curl_es_from_kibana $kpod $eshost '/project.$proj.*/_count' ''"
-        fi
-    done
+    os::log::info See if user $user $negverb read alias /app
+    
+    os::log::info Checking access directly against ES pod...
+    test_message="$user $verb access /app index alias from es"
+    curl_es_pod $espod "/app/_count?pretty"
+    ## Access from elasticsearch pod will not go via es-proxy, so need to pass these headers
+    local xfuser='-H x-forwarded-user:'"$user"
+    local xocpns='-H x-ocp-ns:'"$user_project_list"
+    local xfroles='-H x-forwarded-roles:project_user'
+    local xff='-H x-forwarded-for:localhost'
+    os::cmd::expect_success_and_text \
+        "run_curl_cmd $expected_status curl_es_pod_with_token $espod '/app/_count' $test_token -XGET $xff $xfuser $xocpns $xfroles | get_count_from_json" \
+        "^$expected\$"
+
+    os::log::info Checking access from Kibana pod...
+    ## Access from outside elasticsearch pod, e.g. kiana pod, will go via es-proxy, so the headers will be added by es-proxy
+    test_message="$user $verb access /app index alias from kibana"
+    os::cmd::expect_success_and_text \
+        "run_curl_cmd $expected_status curl_es_from_kibana $kpod $eshost '/app/_count' $test_token -XGET | get_count_from_json" \
+        "^$expected\$"
+
+    # make sure no access with incorrect auth
+    # bogus token
+    os::log::info Checking access providing bogus token
+    test_message="invalid access from es with BOGUS token"
+    os::cmd::expect_success "run_curl_cmd 401 curl_es_pod_with_token $espod '/app/_count' BOGUS"
+
+    test_message="invalid access from kibana with BOGUS token"
+    os::cmd::expect_success "run_curl_cmd 500 curl_es_from_kibana $kpod $eshost '/app/_count' BOGUS"
+
+    # no token
+    os::log::info Checking access providing no username or token
+    test_message="invalid access from es with no token"
+    os::cmd::expect_success "run_curl_cmd 401 curl_es_pod_with_token $espod '/app/_count' ''"
+
+    test_message="invalid access from kibana with no token"
+    os::cmd::expect_success "run_curl_cmd 200 curl_es_from_kibana $kpod $eshost '/app/_count' ''"
 
     test_message="$user has improper access to .operations.* indices from es"
     os::cmd::expect_success_and_text \
-        "run_curl_cmd 403 curl_es_pod_with_token $esopspod '/.operations.*/_count' $test_token | get_count_from_json" \
+        "run_curl_cmd 200 curl_es_pod_with_token $esopspod '/.operations.*/_count' $test_token -XGET $xff $xfuser $xocpns $xfroles| get_count_from_json" \
         "^0\$"
 
     esopshost=$( get_es_svc es-ops )
     if [ "$espod" = "$esopspod" ] ; then
         esopshost=$( get_es_svc es )
     fi
+
+    # accessing from kibana pod will result in 200 with access to 0 docs
     test_message="$user has improper access to .operations.* indices from kibana"
```

Comment:
This and the test messages below still refer to the old `operations.*` indices.

Reply:
need to clarify